packages/core/src/components/Dropdown/Dropdown.tsx

```diff
@@ -18,16 +17,284 @@ import {
+import { Option } from "react-select/src/filters";
+import { HtmlAttributes } from "csstype";
+import { RefObject } from "preact";
+
+interface CustomSingleValueProps extends SingleValueProps<DropdownDefaultValue> {
+  Renderer: React.ComponentType;
+  readOnly: boolean;
+  selectedOption: DropdownDefaultValue;
+  singleVueWrapperClassName?: string;
+}
+
+interface CustomMenuProps extends MenuProps<Option, boolean> {
+  Renderer: React.ComponentType;
+  dropdownMenuWrapperClassName: string;
+}
+
+interface CustomOptionProps extends OptionProps<Option, boolean> {
+  Renderer: React.ComponentType;
+  optionWrapperClassName?: string;
+}
+
+type SelectEvent = {
+  action: "select-option" | "clear";
+  option?: DropdownDefaultValue;
+};
+
+type DropdownState = {
+  isDisabled: boolean;
+  selectProps: { withReadOnlyStyle: boolean; readOnly: boolean };
+};
+
+export interface DropdownComponentProps extends VibeComponentProps {
+  /**
+   * Custom style
+   */
+  className?: string;
+  /** ClassName to be added to dropdown option wrapper (dropdown-wrapper__option--reset) */
+  optionWrapperClassName?: string;
+  /** ClassName to be added to dropdown single value wrapper (dropdown-wrapper__single-value--reset) */
+  singleValueWrapperClassName?: string;
+  /** ClassName to be added to dropdown menu wrapper (dropdown-menu-wrapper) */
+  dropdownMenuWrapperClassName?: string;
+  /**
+   * Placeholder to show when no value was selected
+   */
+  placeholder?: string;
+  /**
+   * If set to true, dropdown will be disabled
+   */
+  disabled?: boolean;
+  /**
+   * If set to true, dropdown won't be editable
+   */
+  readOnly?: boolean;
+  /**
+   * Called when menu is opened
+   */
+  onMenuOpen?: () => void;
+  /**
+   * Called when menu is closed
+   */
+  onMenuClose?: () => void;
+  /**
+   * Called when key is pressed in the dropdown
+   */
+  onKeyDown?: (...args: unknown[]) => unknown;
+  /**
+   * Called when focused
+   */
+  onFocus?: (...args: unknown[]) => unknown;
+  /**
+   * Called when blurred
+   */
+  onBlur?: (...args: unknown[]) => unknown;
+  /**
+   * Called when selected value has changed
+   */
+  onChange?: (...args: unknown[]) => unknown;
+  /**
+   * Called when the dropdown's input changes.
+   */
+  onInputChange?: (...args: unknown[]) => unknown;
+  /**
+   * If true, search in options will be enabled
+   */
+  searchable?: boolean;
+  /**
+   * The dropdown options
+   */
+  options?: DropdownDefaultValue[];
+  /**
+   * Text to display when there are no options
+   */
+  noOptionsMessage?: (() => string) | typeof NOOP;
+  /**
+   * If set to true, the menu will open when focused
+   */
+  openMenuOnFocus?: boolean;
+  /**
+   * If set to true, the menu will open when clicked
+   */
+  openMenuOnClick?: boolean;
+  /**
+   * If set to true, clear button will be added
+   */
+  clearable?: boolean;
+  /**
+   * custom option render function
+   */
+  optionRenderer?: (option: IComboboxOption) => JSX.Element;
```

Comment:
Why using the ComboboxOption?

Reply:
copyed from combobox file, changed to DropdownOption, i assume it should be dropdown option